docs/FeatureManagement/Clients/AspNetCore/Microsoft.FeatureManagement.AspNetCore.md

```diff
@@ -0,0 +1,381 @@
+
+```
+  [FilterAlias("Browser")]
+  public class BrowserFilter : IFeatureFilter
+  {
+      ... Removed for example
+
+      public bool Evaluate(FeatureFilterEvaluationContext context)
+      {
+          BrowserFilterSettings settings = context.Parameters.Get<BrowserFilterSettings>() ?? new BrowserFilterSettings();
+
+          //
+          // Here we would use the settings and see if the request was sent from any of BrowserFilterSettings.AllowedBrowsers
+      }
+  }
+```
+
+### Filter Alias Attribute
+
+When a feature filter is registered to be used for a feature flag, the alias used in configuration is the name of the feature filter type with the _filter_ suffix, if any, removed. For example `MyCriteriaFilter` would be referred to as _MyCriteria_ in configuration.
+
+  "MyFeature": {
+    "EnabledFor": [
+      {
+        "Name": "MyCriteria"
+      }
+    ]
+  }
+
+This can be overridden through the use of the `FilterAliasAttribute`. A feature filter can be decorated with this attribute to declare the name that should be used in configuration to reference this feature filter within a feature flag.
+
+### Using HttpContext
+
+Feature filters can evaluate whether a feature should be enabled based off the properties of an HTTP Request. This is performed by inspecting the HTTP Context. A feature filter can get a reference to the HTTP Context by obtaining an `IHttpContextAccessor` through dependency injection.
+
+```
+public class BrowserFilter : IFeatureFilter
+{
+    private readonly IHttpContextAccessor _httpContextAccessor;
+
+    public BrowserFilter(IHttpContextAccessor httpContextAccessor)
+    {
+        _httpContextAccessor = httpContextAccessor ?? throw new ArgumentNullException(nameof(httpContextAccessor));
+    }
+}
+```
+
+The `IHttpContextAccessor` must be added to the dependency injection container on startup for it to be available. It can be registered in the `IServiceCollection` using the following method.
+
+```
+public void ConfigureServices(IServiceCollection services)
+{
+    …
+    services.TryAddSingleton<IHttpContextAccessor, HttpContextAccessor>();
+    …
+}
+```
+
+### Built-In Feature Filters
+
+There a few feature filters that come with the `Microsoft.FeatureManagement` package. These feature filters are not added automatically, but can be referenced and registered as soon as the package is registered.
+
+Each of the built-in feature filters have their own parameters. Here is the list of feature filters along with examples.
+
+#### Microsoft.Percentage
+
+This filter provides the capability to enable a feature based on a random percentage. If used within ASP.NET Core the `IFeatureManagerSnapshot` can be used to keep the evaluation consistent across a request.
```

Comment:
You have a section for snapshot. It's a bit odd to see the second sentence here.

Reply:
Makes sense. Removed.